modules/csvparser/csvparser.c

```diff
@@ -259,7 +324,8 @@ csv_parser_clone(LogPipe *s)
   cloned->columns = g_list_copy_deep(self->columns, (GCopyFunc) csv_parser_column_clone, NULL);
   csv_scanner_options_copy(&cloned->options, &self->options);
   csv_parser_set_prefix(&cloned->super, self->prefix);
-  csv_parser_set_drop_invalid(&cloned->super, self->drop_invalid);
+  csv_parser_set_drop_invalid(&cloned->super, csv_parser_get_drop_invalid((LogParser *) self));
+  cloned->on_error = self->on_error;
```

Comment:
As the `drop-invalid()` option only operates on the `self->on_error` field, I think it is enough to copy the `on_error` field to the clone.

If we do it like this, we can remove the `csv_parser_get_drop_invalid()` fn, which is only used here.